backend/src/models/config.ts

```diff
@@ -0,0 +1,60 @@
+import { Model, Schema, model } from 'mongoose';
+import Joi from 'joi';
+
+export interface ConfigType {
+  model: {
+    chat?: string;
+    analysis?: string;
```

Comment:
This change making analysis an optional property is what broke the entryConversation unit tests. It wasn't merged into that branch yet which is why it wasn't caught until now. I pushed a commit that fixes the failing tests.

Reply:
Ah I see. Thanks for taking a look! The unit tests pass now. Nice job!